framework/assets/yii.validation.js

```diff
@@ -273,7 +273,12 @@ yii.validation = (function ($) {
             if (options.compareAttribute === undefined) {
                 compareValue = options.compareValue;
             } else {
-                compareValue = $('#' + options.compareAttribute).val();
+                var attributes = $form.data('yiiActiveForm').attributes
+                for (var i = attributes.length - 1; i >= 0; i--) {
```

Comment:
1st speed (do not need to count every time, just simple substraction)
2nd compatible with current version, which takes first element's value